src/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,162 @@
+/* eslint-disable object-curly-newline */
+import { Component } from 'react';
+import './Carousel.scss';
+
+type Props = {
+  images: string[],
+  step: number,
+  frameSize: number,
+  itemWidth: number,
+  animationDuration: number,
+  isFiniteCarousel: boolean,
+};
+
+type State = {
+  itemWidth: number,
+  frameSize: number,
+  step: number,
+  animationDuration: number,
+  coordX: number,
+};
+
+class Carousel extends Component<Props, State> {
+  state: State = {
+    itemWidth: this.props.itemWidth,
+    frameSize: this.props.frameSize,
+    step: this.props.step,
+    animationDuration: this.props.animationDuration,
+    coordX: 0,
+  };
+
+  swipe = (direction: string) => {
```

Comment:
would be better to type it like
```suggestion
  swipe = (direction: 'prev' | 'next') => {
```